src/mdtest.c

```diff
@@ -1210,129 +1214,141 @@ void summarize_results(int iterations) {
-        for (i = 8; i < tableSize; i++) {
+    for (i = start; i < stop; i++) {
             min = max = all[i];
-            for (j = 0; j < iterations; j++) {
-                if(print_time){
-                  curr = summary_table[j].time[i];
-                }else{
-                  curr = summary_table[j].rate[i];
-                }
-
-                if (min > curr) {
-                    min = curr;
-                }
-                if (max < curr) {
-                    max =  curr;
+            for (k=0; k < size; k++) {
+                for (j = 0; j < iterations; j++) {
+                    curr = all[(k*tableSize*iterations)
+                               + (j*tableSize) + i];
+                    if (min > curr) {
+                        min = curr;
+                    }
+                    if (max < curr) {
+                        max =  curr;
+                    }
+                    sum += curr;
                 }
-                sum += curr;
             }
-            mean = sum / (iterations);
-            for (j = 0; j < iterations; j++) {
-                if(print_time){
-                  curr = summary_table[j].time[i];
-                }else{
-                  curr = summary_table[j].rate[i];
+            mean = sum / (iterations * size);
+            for (k=0; k<size; k++) {
+                for (j = 0; j < iterations; j++) {
+                    var += pow((mean -  all[(k*tableSize*iterations)
+                                            + (j*tableSize) + i]), 2);
                 }
-
-                var += pow((mean -  curr), 2);
             }
-            var = var / (iterations);
+            var = var / (iterations * size);
             sd = sqrt(var);
             switch (i) {
-            case 8: strcpy(access, "Tree creation     :"); break;
-            case 9: strcpy(access, "Tree removal      :"); break;
+            case 0: strcpy(access, "Directory creation        :"); break;
+            case 1: strcpy(access, "Directory stat            :"); break;
+                /* case 2: strcpy(access, "Directory read    :"); break; */
+            case 2: ;                                      break; /* N/A */
+            case 3: strcpy(access, "Directory removal         :"); break;
+            case 4: strcpy(access, "File creation             :"); break;
+            case 5: strcpy(access, "File stat                 :"); break;
+            case 6: strcpy(access, "File read                 :"); break;
+            case 7: strcpy(access, "File removal              :"); break;
             default: strcpy(access, "ERR");                 break;
             }
-            fprintf(out_logfile, "   %s ", access);
-            fprintf(out_logfile, "%14.3f ", max);
-            fprintf(out_logfile, "%14.3f ", min);
-            fprintf(out_logfile, "%14.3f ", mean);
-            fprintf(out_logfile, "%14.3f\n", sd);
-            fflush(out_logfile);
+            if (i != 2) {
```

Comment:
This line here 'if (i != 2) {' is hard to read.  What is 'i' and what is '2'?  This is not really related to your PR but it'd be nice if 'i' had a better variable name and if 2 was an enum or something.  Maybe just a comment here to remind the reader what these values are?  Then later we should go back I think and give 'i' a more meaningful name.  'i' is useful as the index within a tight loop but here I think it is being used as something else.